### ISSUE TITLE
Centralised PyTorch 2.0 issues

### ISSUE THREAD

**Issue by Adel-Moumen** [speechbrain/speechbrain#1897]
### Describe the bug

This issue serves as a centralized platform for documenting any bugs or issues encountered when utilizing SpeechBrain in conjunction with PyTorch 2.0.

### Expected behaviour

x

**Comment by jiaweiru** [comment]
### Describe the bug

A small bug: when I use torch 2.0 instead of 1.13.1, DDP training does not work properly.

In general (in version 1.13.1 of torch), running a similar script can help train the model using DDP.
`CUDA_VISIBLE_DEVICES=2,3 python -m torch.distributed.launch --nproc_per_node=2 train_hifigan.py hparams/train_hifigan.yaml --distributed_launch --distributed_backend=nccl`

But in torch 2.0, this does not seem to be available. It seems to be caused by an update of torch, which can be referred to [this.](https://github.com/pytorch/pytorch/commit/a229b4526f41eed34a871e488f5c86d43f63d013)
The content in this commit is to change underscores to hyphens in some parameter names in the parser. Although this commit has little impact on the native use of torch because it also supports underscore parameters, in SpeechBrain, this will cause an error because `speechbrain.parse_arguments()` can only recognize the `--local_rank` provided by the previous version of torch.distributed.launch, but not the `--local-rank` in version 2.0. Therefore, parse_arguments() will recognize `--local-rank` as an override and attempt to overwrite a non-existent local-ralocal-ranknk parameter in the hparams.

### Expected behaviour

xxxx

### To Reproduce
In my experiment, it will prompt an error that looks like this.

```
Traceback (most recent call last):
    File "/home/sturjw/Code/TTSS/train_hifigan.py", line 336, in <module>
        hparams = load_hyperpyyaml(fin, overrides)
    File "/home/sturjw/.conda/envs/SB/lib/python3.9/site-packages/hyperpyyaml/core.py", line 157, in load_hyperpyyaml
        yaml_stream = resolve_references(yaml_stream, overrides, overrides_must_match)
    File "/home/sturjw/.conda/envs/SB/lib/python3.9/site-packages/hyperpyyaml/core.py", line 316, in resolve_references
        recursive_update(preview, overrides, must_match=overrides_must_match)
    File "/home/sturjw/.conda/envs/SB/lib/python3.9/site-packages/hyperpyyaml/core.py", line 776, in recursive_update
        raise KeyError(f"Override '{k}' not found in: {[key for key in d.keys()]}")
KeyError: "Override 'local-rank' not found in: ['seed', '__set_seed', 'output_folder', 'save_folder', 'train_log', 'progress_sample_path', 'epochs', 'keep_checkpoint_interval', 'use_tensorboard', 'data_folder', 'libritts_subsets', 'json_folder', 'train_json', 'valid_json', 'test_json', 'skip_prep', 'splits', 'sheckpoint_interval', 'use_tensorboard', 'data_folder', 'libritts_subsets', 'json_folder', 'train_json', 'valid_json', 'test_json', 'skip_prep', 'splits', 'split_ratio', 'segment_size', 'sample_rate', 'hop_length', 'win_length', 'n_mel_channels', 'n_fft', 'mel_fmin', 'mel_fmax', 'mel_normalized', 'power', 'norm'.....]
```
As I mentioned above, the error can be traced back to attempting to overwrite a parameter that does not exist in the YAML file.

Therefore, I believe that this is not a specific use case, and this bug should be reproducible in any attempt to use DDP to train models in Torch 2.0.

So this situation should be easily reproducible by simply installing Torch 2.0.

### Versions

torch 2.0
speechbrain 0.5.14

### Relevant log output

No response

### Additional context

I ultimately resolved this issue by rolling back the Torch version.

But I believe that this bug can be easily fixed and made compatible both with Torch 1.x and 2.0 by changing `parser.add_argument("--local_rank", type=int, help="Rank on local machine")` to `parser.add_argument("--local_rank", "--local-rank", type=int, help="Rank on local machine")` in `speechbrain.parse_arguments()`.

Also, I am not a native English speaker, so if my expressions confuse you, please ask me. Finally, I hope this will help you make SB better.

**Comment by Adel-Moumen** [comment]
Hi @jiaweiru, thanks for reporting this issue. 

This has been fixed in #1951. 

Many thanks for spotting this issue and making SB better. (PS: your English is really good don't worry :))

**Comment by accentmagic** [comment]
When I tried upgrading PyTorch to 2.0,  I started getting the following error from Checkpointer while moving from the training to validation stage (first time it attempts to save the checkpoint):


  File "C:\.......\speechbrain\speechbrain\utils\checkpoints.py", line 572, in save_checkpoint
    raise RuntimeError(MSG)
RuntimeError: Don't know how to save <class 'torch.optim.lr_scheduler.ExponentialLR'>. Register default hook or add custom hook for this object.
